File: src/frontend/lib/patienten/infrastructure/patient_form_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'generated/patient_form_dto.freezed.dart';
part 'generated/patient_form_dto.g.dart';

@freezed
abstract class PatientFormDto with _$PatientFormDto {
  const factory PatientFormDto({
    required String vorname,
    required String nachname,
    required DateTime geburtstag,
    String? titel,
    String? strasse,
    String? hausnummer,
    String? plz,
    String? stadt,
    String? telMobil,
    String? telFestnetz,
    String? email,
  }) = _PatientFormDto;

  const PatientFormDto._();

  factory PatientFormDto.fromJson(Map<String, dynamic> json) => _$PatientFormDtoFromJson(json);
}

File: src/main/kotlin/de/keller/physioai/patienten/adapters/api/PatientFormDto.kt
package de.keller.physioai.patienten.adapters.api

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.Past
import org.springframework.format.annotation.DateTimeFormat
import java.time.LocalDate

data class PatientFormDto(
    val titel: String?,
    val vorname: String?,
    val nachname: String?,
    val strasse: String?,
    val hausnummer: String?,
    val plz: String?,
    val stadt: String?,
    val telMobil: String?,
    val telFestnetz: String?,
    @Email
    val email: String?,
    @Past
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    val geburtstag: LocalDate?,
)
